### PR TITLE
Add support for resource memory limit to generate kube

### PR DESCRIPTION
**Pull request by andylibrian**
Addresses #7855

Steps to run:
```
podman container create --name test1 --memory 10Mi alpine
podman generate kube test1
```

Output (see the comment / note in the generated yaml):
```
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: "2020-10-04T14:20:48Z"
  labels:
    app: test1
  name: test1
spec:
  containers:
  - command:
    - /bin/sh
    env:
    - name: PATH
      value: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
    - name: TERM
      value: xterm
    - name: container
      value: podman
    - name: HOSTNAME
    image: docker.io/library/alpine:latest
    name: test1
    resources:
      limits:
        memory: 10Mi          # <<<-- This PR adds this
```

### CODE REVIEW

**Comment by rhatdan** [comment]
/approve
@haircommander @vrothberg @saschagrunert @zhangguanzhang @mheon PTAL

File: libpod/kube.go [hunk 1]
```diff
@@ -303,12 +303,23 @@ func containerToV1Container(c *Container) (v1.Container, []v1.Volume, error) {
 	// This should not be applicable
 	//container.EnvFromSource =
 	kubeContainer.Env = envVariables
-	// TODO enable resources when we can support naming conventions
-	//container.Resources
 	kubeContainer.SecurityContext = kubeSec
 	kubeContainer.StdinOnce = false
 	kubeContainer.TTY = c.config.Spec.Process.Terminal
 
+	if c.config.Spec.Linux != nil &&
```
Comment (mheon) [inline libpod/kube.go:310]: Can you add a TODO for adding CPU limiting support?

Comment (andylibrian) [reply libpod/kube.go:311]: Added. Thanks @mheon.

**Review by saschagrunert — APPROVED** [review]
LGTM

**Pull request merged**